src/frontend/src/views/Data/components/SamplesView/components/SamplesTable/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { CellBasic, CellHeader, Table, TableHeader, TableRow } from "czifui";
+import {
+  createColumnHelper,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { IdMap } from "src/common/utils/dataTransforms";
+import { map } from "lodash";
+import { useEffect, useState } from "react";
+
+// TODO-TR (mlila): types
+interface Props {
+  data: IdMap<Sample> | undefined;
+  isLoading: boolean;
+}
+
+const columnHelper = createColumnHelper<Sample>();
+
+const columns = [
+  columnHelper.accessor("privateId", {
+    header: "Private ID",
+  }),
+  columnHelper.accessor("publicId", {
+    header: "Public ID",
+  }),
+  columnHelper.accessor("collectionDate", {
+    header: "Collection Date",
+  }),
+  columnHelper.accessor((obj) => JSON.stringify(obj.lineage), {
```

Comment:
I assume the `(obj) => JSON.stringify(obj.foo)` stuff here and below is mostly about proof of concept to see the data is available and being passed to the table? Any plans yet on passing the more complicated data chunks downstream to the cells (eg, passing down the full `lineage` to those cells so they can do the full data in their tooltip)? This was a pain for me in the past, but I did with a combination of custom accessors and then custom cell components that were specialized to those outputs. I'm looking over the current react-table docs, that looks doable, although how you call the specified `cell` component seems confusing to me at the moment.

Reply:
yes exactly. this is the most basic table just to prove we can render the thing